activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -88,31 +88,15 @@ def alias_attribute_method_definition(code_generator, pattern, new_name, old_nam
         parameters = pattern.parameters
         old_name = old_name.to_s
 
-        method_defined = method_defined?(target_name) || private_method_defined?(target_name)
-        manually_defined = method_defined &&
-          !self.instance_method(target_name).owner.is_a?(GeneratedAttributeMethods)
-        reserved_method_name = ::ActiveRecord::AttributeMethods.dangerous_attribute_methods.include?(target_name)
-
         if !abstract_class? && !has_attribute?(old_name)
-          # We only need to issue this deprecation warning once, so we issue it when defining the original reader method.
-          should_warn = target_name == old_name
-          if should_warn
-            ActiveRecord.deprecator.warn(
-              "#{self} model aliases `#{old_name}`, but `#{old_name}` is not an attribute. " \
-              "Starting in Rails 7.2, alias_attribute with non-attribute targets will raise. " \
+          # We only need to raise once, so we issue it when defining the original reader method.
+          is_reader_method = target_name == old_name
+
+          if is_reader_method
```

Comment:
We don't need this condition anymore. We needed it to avoid raising more than one deprecation warning but with `raise` it  shouldn't matter since `raise` will halt further method definitions